lib/main.js

```diff
@@ -0,0 +1,103 @@
+'use babel'
+
+import { CompositeDisposable } from 'atom';
+import { Linterhub, LinterhubTypes } from 'linterhub-ide';
+import { Integration } from './integration';
+import * as path from 'path';
+import LinterhubStatusView from './views/status';
+import LinterhubManagerView from './views/manager';
+
+let subscriptions
+let statusBar
+let managerView
+let linter
+
+export default {
+  config: {
+    'linterhub-atom.mode': {
+      'description': 'How to run linters',
+      'type': 'integer',
+      'default': null,
+    },
+    'linterhub-atom.path_cli': {
+      'description': 'Path to cli',
+      'type': 'string',
+      'default': null,
+    }
+  }
+};
+
+function show_version() {
+  Linterhub.version().then((hubVersion) => {
+    atom.notifications.addSuccess(hubVersion);
+  });
+}
+
+function show_manager() {
+  if(managerView == null)
+    managerView = new LinterhubManagerView(subscriptions);
+  managerView.show();
+}
+
+export function activate() {
+
+  subscriptions = new CompositeDisposable();
+  subscriptions.add(atom.commands.add('atom-workspace', {
+    'linterhub-atom:version': () => show_version(),
+    'linterhub-atom:manager': () => show_manager()
+  }));
+}
+export function consumeSignal(registry) {
+  statusBar = registry.create()
+  subscriptions.add(statusBar)
+  subscriptions.add(
+    atom.config.observe('linterhub-atom.status', (value) => {
+      if (value !== undefined) {
+        statusBar.clear();
+        if(value != "Active")
+          statusBar.add(value);
+      }
+    })
+  );
+  subscriptions.add(
+    atom.config.observe('linterhub-atom.progress', (value) => {
+      if (value !== undefined) {
+        if (!value) {
+          statusBar.clear();
+        }
+      }
+    })
+  );
+}
+
+export function deactivate() {
+  subscriptions.dispose()
+}
+
+export function consumeIndie(registerIndie) {
+  linter = registerIndie({
+    name: 'Linterhub',
+  })
+
+  let settings = {
+    linterhub: {
+      enable: true,
+      mode: atom.config.get('linterhub-atom.mode'),
```

Comment:
You can `atom.config.observe` these keys in your `activate()`, that way you aren't incurring the cost of determining your settings value every time this gets called since it will set it on activate and update whenever the setting is changed.